staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -234,6 +234,23 @@ type Store struct {
 	// If set, DestroyFunc has to be implemented in thread-safe way and
 	// be prepared for being called more than once.
 	DestroyFunc func()
+
+	// CorruptObjDeleter, if initialized, will make an attempt to
+	// perform the normal deletion flow, but if either of the below occurs:
+	// a) the data (associated with the resource being deleted) retrieved
+	// from the storage failed to transform properly (eg. decryption failure)
+	// b) the data (associated with the resource being deleted) failed to
+	// decode properly (eg. corrupt data)
+	// it will disregard these errors, bypass the finalzer constraints,
+	// deletion hook(s) and go ahead with the deletion of the object.
+	//
+	// WARNING: This may break the cluster if the resource has
+	// dependencies. Use when the cluster is broken, and there is no
+	// other viable option to repair the cluster.
+	//
+	// TODO: it's located here for convenience so we can wire it
+	// from inside the handler.DeleteResource function
```

Comment:
what is the TODO? Is it wrong here? Where should it go?

Reply:
ideally i would like to do wrap the `registry.Store` with the following:
```
type corruptObjectDeleter struct {
    *registry.Store
}
```

and pass an instance of `corruptObjectDeleter` to api installer, but REST factory functions for all types directly instantiates `registry.Store{...}`

https://github.com/kubernetes/kubernetes/blob/f1e447b9d32ac325074380d239370cde02a6dbf7/pkg/registry/core/endpoint/storage/storage.go#L36-L55

So I am holding the `CorruptObjDeleter` instance inside `registry.Store` so I can access it from 
https://github.com/kubernetes/kubernetes/blob/f1e447b9d32ac325074380d239370cde02a6dbf7/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go#L50